packages/starlight-theme-ensnode/starlight.ts

```diff
@@ -0,0 +1,39 @@
+import type { HookParameters } from "@astrojs/starlight/types";
+import type { AstroIntegrationLogger } from "astro";
+
+export function overrideComponents(
+  starlightConfig: StarlightUserConfig,
+  overrides: ComponentOverride[],
+  logger: AstroIntegrationLogger
+): StarlightUserConfig["components"] {
+  const components = { ...starlightConfig.components };
+  for (const override of overrides) {
+    const name = typeof override === "string" ? override : override.name;
+
+    if (starlightConfig.components?.[name]) {
+      const fallback = `starlight-theme-ensnode/${
+        typeof override === "string" ? "overrides" : "components"
+      }/${typeof override === "string" ? override : override.fallback}.astro`;
```

Comment:
How does that alternative form look? Was aiming to make it easier reason about.
```suggestion
      const fallback = typeof override === "string"
        ? `starlight-theme-ensnode/overrides/${override}.astro`
        : `starlight-theme-ensnode/components/${override.fallback}.astro`;
```